src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/TagBuilder.cs

```diff
@@ -195,56 +177,44 @@ public void MergeAttribute(string key, string value, bool replaceExisting)
 
         public void SetInnerText(string innerText)
         {
-            InnerHtml = _htmlEncoder.HtmlEncode(innerText);
-        }
-
-        public HtmlString ToHtmlString(TagRenderMode renderMode)
-        {
-            return new HtmlString(ToString(renderMode));
+            InnerHtml = new StringHtmlContent(innerText);
         }
```

Comment:
That would mean `ToString`ing wherever `InnerHtml` is used. Example - <a href="https://github.com/aspnet/Mvc/pull/2899/files#diff-0e83439eeed8ffbde4eceb0db1f6e541L721">Here</a> `BufferedHtmlContent` is set as the `TagBuilder.InnerHtml`. So `SetInnerText` acts like a shortcut when you have un-encoded text to be set as the content for the TagBuilder.